metaci/plan/models.py

```diff
@@ -191,9 +201,19 @@ def check_github_event(self, event, payload):
             and self.trigger == "status"
             and payload["state"] == "success"
         ):
-            if not re.match(self.regex, payload["context"]):
+            if not re.match(self.commit_status_regex, payload["context"]):
                 return run_build, commit, commit_message
 
+            # If we also have a branch regex filter, run it.
+            if self.regex:
+                if not payload["ref"].startswith("refs/heads/"):
+                    return run_build, commit, commit_message
+
+                branch = payload["ref"][11:]
+                # Check the branch against regex
+                if not re.match(self.regex, branch):
+                    return run_build, commit, commit_message
```

Comment:
Maybe put this in a helper function used by both the trigger="commit" and trigger="status" blocks, instead of duplicating?

Reply:
I did, but then realized that the structure this is looking for doesn't actually match between the status and commit webhook payloads. Refactoring.

Reply:
Good catch